javalin/src/main/java/io/javalin/config/EventConfig.kt

```diff
@@ -50,6 +50,15 @@ class EventConfig(private val cfg: JavalinConfig) {
         eventManager.wsHandlerAddedHandlers.add(callback);
     }
 
+    /**
+     * Registers a handler for a user-defined event type.
+     * @param eventClass the class of the event
+     * @param handler the handler to be called when the event is emitted
+     */
+    fun <T : Any> on(eventClass: Class<T>, handler: Consumer<T>) {
```

Comment:
I'd call it `listener` and maybe even require to use our own `Listener` interface (in the future we can mount some additional parameters on it, like e.g. priority)